scipy/stats/_distribution_infrastructure.py

```diff
@@ -2034,8 +2111,10 @@ def f2(x, _p, **kwargs):  # named `_p` to avoid conflict with shape `p`
         res = _bracket_root(f3, xl0=a, xr0=b, xmin=xmin, xmax=xmax, args=args)
         # For now, we ignore the status, but I want to use the bracket width
         # as an error estimate - see question 5 at the top.
+
         xrtol = None if _isnull(self.tol) else self.tol
-        return _chandrupatla(f3, a=res.xl, b=res.xr, args=args, xrtol=xrtol).x
+        xatol = None if xatol is None else xatol
+        return _chandrupatla(f3, a=res.xl, b=res.xr, args=args, xrtol=xrtol, xatol=xatol)
```

Comment:
Change 3: when inverting a discrete distribution function numerically, set `xatol=1` to avoid unnecessary iteration. The integer root is the only integer within the bracket. (Special cases - if the bracket was not valid or if the final bracket happens to have integer endpoints - are handled in the functions that call `_solve_bounded`. The right thing to do depends on whether it's the CDF or CCDF.)

Note that this now returns the whole result object so the caller method can see the result bracket and function values. This is why we had to change the `if not p.size` case above, and in the continuous inverse methods, we now have to extract attribute `x`.